guessTheNumber.js

```diff
@@ -1,17 +1,42 @@
+const readline = require('readline');
 // ### 1: Guess The Number
 //
 // Write a program that will generate a number between 1 and 100. Your program will ask the user to guess the generated number.
 //
 // When the user types a guess, the program should tell them if they won (the guess matched the computer's number), if their guess was higher than the number the computer generated, OR if the guess was lower than the number the computer generated. Terminate the program if the user types "exit".
 
-  var compNum = []
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+})
 
-var numGenerator = function() {
-  while(compNum.length < 1){
-    var randomNumber = Math.ceil(Math.random()*100)
-    if(compNum.indexOf(randomNumber) > -1) continue;
-    compNum[compNum.length] = randomNumber;
+
+function numGenerator( min, max ) {
+  min = Math.ceil(min);
```

Comment:
If we needed these temporary variables, I would create local copies to store these values.  I try to avoid writing code that modifies the parameters passed in, to avoid unanticipated side effects for the caller.  (See https://en.wikipedia.org/wiki/Pure_function for more information.)